alma/views/js/alma-inpage.js

```diff
@@ -22,215 +22,157 @@
+                    blockForm.hidden = true;
+                }
+                let url = formInpage.dataset.action;
 
-                    window.location.href = urlError;
-                });
-        }
-    }
+                inPage = createIframe(formInpage);
 
-    const isAlmaPayment = function (url) {
-        return url.indexOf("module/alma/payment") !== -1 || url.indexOf("module=alma") !== -1;
-    };
+                const eventAlma = async function (e) {
+                    e.preventDefault();
+                    e.stopPropagation();
+                    await createPayment(url, inPage);
+                };
 
-    const getURLParameter = function(sUrl, sParam)
-    {
-        var sURLVariables = sUrl.split('?');
-        for (var i = 0; i < sURLVariables.length; i++)
-        {
-            var sParameterName = sURLVariables[i].split('=');
-            if (sParameterName[0] == sParam)
-            {
-                return decodeURIComponent(sParameterName[1]);
+                paymentButtonEvents.push(eventAlma);
+                paymentButton.addEventListener('click', eventAlma);
             }
-        }
-    };
+        });
+    });
 
-    const createIframe = function (paymentOptionId, selectorSetting, showPayButton, url = '') {
-        let merchantId = selectorSetting.data('merchantid');
-        let installment = selectorSetting.data('installment');
-        let purchaseAmount = selectorSetting.data('purchaseamount');
-        let locale = selectorSetting.data('locale');
+    //Prestashop 1.6-
+    let paymentButtonsPs16 = document.querySelectorAll(".alma-inpage.ps16");
+    paymentButtonsPs16.forEach(function (button) {
+        button.addEventListener('click', function (e) {
+            e.preventDefault();
+            let paymentOptionId = this.getAttribute('id');
+            let url = this.getAttribute('href');
+            let settingInpage = document.querySelector('#alma-inpage-' + paymentOptionId);
 
-        selectorSetting.attr('id', 'alma-inpage-' + paymentOptionId);
+            if( inPage !== undefined) {
+                inPage.unmount();
+            }
 
-        let selectorIframeInPage = '#alma-inpage-' + paymentOptionId;
+            createIframe(settingInpage, true, url);
+        });
+    });
+}
 
-        if (showPayButton) {
-            // No refactor inPage is use in callback function 1.6
-            inPage = Alma.InPage.initialize(
-                {
-                    merchantId: merchantId,
-                    amountInCents: purchaseAmount,
-                    installmentsCount: installment,
-                    locale: locale,
-                    environment: selectorSetting.data("apimode"),
-                    selector: selectorIframeInPage,
-                    onIntegratedPayButtonClicked : () => {
-                        ajaxPayment(url, inPage);
-                    }
-                }
-            );
+function createIframe(form, showPayButton = false, url = '') {
```

Comment:
I assume we could have function name conflicts, depending on the modules loaded by the CMS.

```suggestion
function createAlmaIframe(form, showPayButton = false, url = '') {
```